site/journal/2020/Understanding args in Glimmer Components.md

```diff
@@ -0,0 +1,309 @@
+---
+title: Understanding `args` in Glimmer Components
+subtitle: Clearing up a common confusion with a worked example.
+date: 2020-12-21T09:00
+summary: >
+  Many developer assume this is more magic going on with Glimmer components’ arguments than there really is. Let’s see how they actually work!
+image:
+qualifiers:
+  audience: >
+    Software developers working with Ember Octane; also developers with a decent amount of JavaScript knowledge interested in deepening their understanding of modern JavaScript.
+
+thanks: >
+  A conversation with [Nathaniel Furniss](https://www.linkedin.com/in/nlfurniss/) helped me see the value of stripping out all the autotracking and Ember/Glimmer-isms for explaining this concept. [Chris Garrett](https://pzuraq.com) ([@pzuraq](https://github.com/pzuraq)) pointed me to the actual Glimmer implementation of the concepts discussed in this post.
+
+tags:
+  - software development
+  - Ember
+  - JavaScript
+  - mental models
+
+templateEngineOverride: md
+
+---
+
+One of the most consistent confusions I see as I support the LinkedIn.com migration to [Ember Octane](https://emberjs.com/editions/octane/) is how to think about `args` in Glimmer components. In particular, I consistently see people struggling with how to understand *updates* to `args` what they can and cannot do with them. In this post, I hope to make it much clearer by working through an example implementation of a slightly simpler version of the `Component` API, with no autotracking in sight. By the end of the post, you should have a clear handle on what will and won’t work in the body of a Glimmer component—and, more importantly, *why*.
```

Comment:
> I hope to make it much clearer by working through an example implementation of a slightly simpler version of the `Component` API, with no autotracking in sight

I feel like the framing here is a bit off. Maybe something like:

> an plain JS example implementation of the way that arguments are passed from component to component in Glimmer

You're really focusing on one aspect of how components work, which is how arguments are passed down and accessed. There's a lot more to the Component API as a whole - not the TS interface of course, but like, all of the things that make up a component in general to the average user

Reply:
Ah, I like that tweak and clarification—this is really just focused on args and the JS API people interact with. Thank you!